### PR TITLE
chore(tracing): improve benchmarks for span creation

### PR DESCRIPTION
**Pull request by mabdinur**
Currently span creation benchmark measure the duration of `Span.__init__(....)`. I don't think this is meaningful since using `Span.__init__()` accounts for a small fraction of span creation. 

Instead we should benchmark the duration of creating a span using `tracer.trace()`. This will add the duration of configuring sampling configurations, parenting spans, and running default span finish hooks to our benchmarks. This will paint a more accurate picture of the overhead of generating spans.

New benchmarks
```
.....................
span-start: Mean +- std dev: 6.50 ms +- 0.53 ms
.....................
span-add-tags: Mean +- std dev: 55.2 ms +- 1.5 ms
.....................
span-add-metrics: Mean +- std dev: 35.1 ms +- 1.0 ms
.....................
span-start-finish: Mean +- std dev: 39.6 ms +- 0.6 ms
```

Previous Benchmarks
```
.....................
span-start: Mean +- std dev: 944 us +- 20 us
.....................
span-add-tags: Mean +- std dev: 52.1 ms +- 0.6 ms
.....................
span-add-metrics: Mean +- std dev: 33.4 ms +- 0.5 ms
.....................
span-start-finish: Mean +- std dev: 1.19 ms +- 0.02 ms
```

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] PR description includes explicit acknowledgement/acceptance of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist

- [ ] Title is accurate.
- [ ] No unnecessary changes are introduced.
- [ ] Description motivates each change.
- [ ] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Testing strategy adequately addresses listed risk(s).
- [ ] Change is maintainable (easy to change, telemetry, documentation).
- [ ] Release note makes sense to a user of the library.
- [ ] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: benchmarks/span/scenario.py [hunk 1]
```diff
@@ -1,6 +1,12 @@
 import bm
 import bm.utils as utils
 
+from ddtrace import tracer
+
+
+# To reduce noise avoid setting finished spans to the agent. Instead write encoded spans to stdout
```
Comment (mabdinur) [inline benchmarks/span/scenario.py:7]:
```suggestion
# To reduce noise avoid sending finished spans to the agent
```

File: benchmarks/span/scenario.py [hunk 2]
```diff
@@ -24,7 +30,7 @@ def run(self):
         def _(loops):
             for _ in range(loops):
                 for i in range(self.nspans):
-                    s = utils.gen_span("test." + str(i))
```
Comment (Yun-Kim) [inline benchmarks/span/scenario.py:27]: We already have a benchmark scenario to cover `tracer.trace()` here: https://github.com/DataDog/dd-trace-py/blob/d4343d340b48e37f5e5f6afa5656ae65043404c5/benchmarks/tracer/scenario.py#L25

Comment (mabdinur) [reply benchmarks/span/scenario.py:27]: ahhh I'll close this PR

**Comment by mabdinur** [comment]
`tracer.trace()` benchmarks already exist

**Pull request closed without merging**